common/lib/xmodule/xmodule/randomize_module.py

```diff
@@ -58,15 +56,23 @@ def __init__(self, *args, **kwargs):
                 else:
                     self.choice = random.randrange(0, num_choices)
 
+        if self.child is not None:
+            log.debug("children of randomize module (should be only 1): %s", self.child)
+
+    @property
+    def child_descriptor(self):
+        """ Return descriptor of selected choice """
         if self.choice is not None:
-            self.child_descriptor = self.descriptor.get_children()[self.choice]
-            # Now get_children() should return a list with one element
-            log.debug("children of randomize module (should be only 1): %s",
-                      self.get_children())
-            self.child = self.get_children()[0]
-        else:
-            self.child_descriptor = None
-            self.child = None
+            return self.descriptor.get_children()[self.choice]
+        return None
```

Comment:
Nit: It might be clearer to say:

    if self.choice is None:
        return None

    return self.descriptor.get_children()[self.choice]